quickwit/quickwit-actors/src/actor_context.rs

```diff
@@ -340,16 +331,13 @@ impl<A: Actor> ActorContext<A> {
         M: Sync + Send + std::fmt::Debug + 'static,
     {
         let self_mailbox = self.inner.self_mailbox.clone();
-        let callback = Callback(Box::pin(async move {
+        let callback = async move {
```

Comment:
Can you add the following docs for this function? This is not obvious that a scheduled message to self is sent into the high-priority queue.  

```
    /// Schedules a message that will be sent to the high-priority
    /// queue of the actor Mailbox.
    /// This function is especially useful for an actor to monitor/observe himself. Using the high-priority queue ensures that the message will be processed after the processing of all queued messages, which will defeat the purpose of monitoring.
```